packages/api/test/e2e/doughnut.e2e.ts

```diff
@@ -0,0 +1,299 @@
+    return d.encode();
+}
+
+describe('Doughnut for CennznetExtrinsic', () => {
+    let aliceKeyPair = {
+        secretKey: hexToU8a('0x98319d4ff8a9508c4bb0cf0b5a78d760a0b2082c02775e6e82370816fedfff48925a225d97aa00682d6a59b95b18780c10d7032336e88f3442b42361f4a66011'),
+        publicKey: hexToU8a('0xd43593c715fdd31c61141abd04a99fd6822c8558854ccde39a5684e7a56da27d'),
+    };
+
+    let api: Api;
+    let alice, bob, charlie, dave;
+    let keyring: {
+        [index: string]: KeyringPair;
+    };
+
+    beforeAll(async () => {
+        api = await initApiPromise();
+        await cryptoWaitReady();
+        const keyring = new Keyring({type: 'sr25519'});
+        alice = keyring.addFromUri('//Alice');
+        bob = keyring.addFromUri('//Bob');
+        charlie = keyring.addFromUri('//Charlie');
+        dave = keyring.addFromUri('//Dave');
+    });
+
+    it('Delegates a GA transfer from alice to charlie when extrinsic is signed by bob', async done => {
+
+        let doughnut = makeDoughnut(
+            aliceKeyPair,
+            bob,
+            {'cennznet': makeCennznut('generic-asset', 'transfer')},
+        );
+        const CENNZ = 16000;
+        const reciever = dave.address;
+        const amount = 8767535;
+
+        const tx = api.tx.genericAsset.transfer(CENNZ, reciever, amount);
+        const doughnutB = new Doughnut(api.registry, doughnut);
+        const opt = {doughnut: doughnutB};
+        await tx.signAndSend(bob, opt as any, async ({events, status}) => {
+            if (status.isFinalized) {
+                const blockHash = status.value.toHex();
+                console.log('Completed at block hash', blockHash);
+                const allEvents = await api.query.system.events.at(blockHash);
+                const transfer = checkEventMatches(
+                    {
+                        allEvents,
+                        method: 'Transferred',
+                        section: 'genericAsset',
+                        sender: alice.address,
+                        reciever,
+                        amount: amount.toString(),
+                        asset: CENNZ.toString(),
+                    });
+                allEvents.forEach(({phase, event: {data, method, section}}) => {
+                    console.log('\t', phase.toString(), `: ${section}.${method}`, data.toString());
+                });
+                if (transfer != undefined) {
+                    done();
+                } else {
+                    assert(true, 'false');
+                }
+            }
+        });
+    });
+
+
+    function checkEventMatches({allEvents, method, section, sender, reciever, amount, asset}) {
+        if (method === 'Transferred') {
+            return allEvents.find(
+                event => (
+                    event.event.data.method === method &&
+                    event.event.data.section === section &&
+                    event.event.data[1].toString() === sender &&
```

Comment:
Wouldn't it be better to remove the check for "sender", so you don't need if (method..)else? If you agree, please remove the sender from the arguments too.

Reply:
as discussed..